### PR TITLE
Additional System.Diagnostics.Contracts tests

### PR DESCRIPTION
**Pull request by stephentoub**
Added tests for all of the Contract.* members that don't require the rewriter. Also updated the existing tests to appropriately unregister from the ContractFailed event, and to disable parallelization for the test assembly as ContractFailed represents static state that will affect other tests running concurrently.

### CODE REVIEW

**Comment by stephentoub** [comment]
cc: @FiveTimesTheFun

**Comment by FiveTimesTheFun** [comment]
Thanks Stephen, looking soon.

File: src/System.Diagnostics.Contracts/tests/System.Diagnostics.Contracts.Tests.csproj [hunk 1]
```diff
@@ -1,10 +1,10 @@
-<?xml version="1.0" encoding="utf-8"?>
+﻿<?xml version="1.0" encoding="utf-8"?>
 <Project ToolsVersion="12.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003" DefaultTargets="Build">
   <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
   <PropertyGroup>
     <Configuration Condition="'$(Configuration)' == ''">Debug</Configuration>
     <Platform Condition="'$(Platform)' == ''">AnyCPU</Platform>
-    <ProjectGuid>{0C4E7FF1-54C6-49b7-9700-18F5F3EB8E65}</ProjectGuid>
+    <ProjectGuid>{0C4E7FF1-54C6-49B7-9700-18F5F3EB8E65}</ProjectGuid>
```
Comment (FiveTimesTheFun) [inline src/System.Diagnostics.Contracts/tests/System.Diagnostics.Contracts.Tests.csproj]: Hi Stephen, did you do this or did VS?  Should I be doing this manually?  I get my guids from guidgen.

File: src/System.Diagnostics.Contracts/tests/ForAllTests.cs [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+{
+    public class ForAllTests
+    {
+        [Fact]
+        public static void ArgumentExceptions()
+        {
+            Assert.Throws<ArgumentNullException>(() => Contract.ForAll(0, 1, null));
+            Assert.Throws<ArgumentNullException>(() => Contract.ForAll<int>(null, i => true));
+            Assert.Throws<ArgumentNullException>(() => Contract.ForAll<int>(Enumerable.Empty<int>(), null));
+            Assert.Throws<ArgumentException>(() => Contract.ForAll(1, 0, i => true));
```
Comment (FiveTimesTheFun) [inline src/System.Diagnostics.Contracts/tests/ForAllTests.cs]: // fromInclusive larger than toExclusive

File: src/System.Diagnostics.Contracts/tests/ForAllTests.cs [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        }
+
+        [Fact]
+        public static void EmptyInputReturnsTrue()
+        {
+            Assert.True(Contract.ForAll(Enumerable.Empty<int>(), i => true));
+            Assert.True(Contract.ForAll(-2, -2, i => true));
+            Assert.True(Contract.ForAll(1, 1, i => true));
```
Comment (FiveTimesTheFun) [inline src/System.Diagnostics.Contracts/tests/ForAllTests.cs]: i => throw new InvalidOperationException("Didn't expect to call predicate") (for all of these)?

**Comment by FiveTimesTheFun** [comment]
Thanks Stephen - a couple comments, but looks good to me.

**Comment by stephentoub** [comment]
Thanks, Matt.

**Pull request merged**